### PR TITLE
refactor(test): merge simulation and simulation_scale

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR merges `src/tests/simulation_scale` to `src/tests/simulation`.

It also adds some arguments to `risingwave_simulation`:
- `--kill`: alias to `--kill-meta --kill-frontend --kill-compute --kill-compactor`
- `--config-path`: specify a configuration file

After this PR, we can start working on testing **scaling + failover**. 😈

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/tests/simulation/src/cluster.rs [hunk 1]
```diff
@@ -12,19 +12,19 @@
...
 use madsim::runtime::{Handle, NodeHandle};
-use rand::seq::SliceRandom;
+use rand::Rng;
+use sqllogictest::AsyncDB;
 
-use crate::RisingWave;
+use crate::client::RisingWave;
 
 /// Embed the config file and create a temporary file at runtime.
 static CONFIG_PATH: LazyLock<tempfile::TempPath> = LazyLock::new(|| {
```
Comment (BugenZhao) [inline src/tests/simulation/src/cluster.rs:30]: Do we want this config for simulation e2e? 🤔

File: Makefile.toml [hunk 2]
```diff
@@ -596,7 +596,7 @@ script = """
 set -e
 
 cargo nextest archive \
-  -p risingwave_simulation_scale \
+  -p risingwave_simulation \
```
Comment (BugenZhao) [inline Makefile.toml:599]: How about moving the scale-related tests to a separate module and only archive that module here? 🤔 Here we rely on that all Rust tests in this package is about scaling.

**Review by BugenZhao — COMMENTED** [review]
🚀🥰

File: Makefile.toml (hunk 2 above)
Comment (wangrunji0408) [reply Makefile.toml:599]: Seems that we can archive all tests here, but only run scale-related tests by specifying the name in `deterministic-scale-test.sh`.

Comment (BugenZhao) [reply Makefile.toml:599]: +1

File: src/tests/simulation/src/cluster.rs (hunk 1 above)
Comment (BugenZhao) [reply src/tests/simulation/src/cluster.rs:30]: The product of these two parameters decides the update frequency of the query results. In scale tests, we rely on the changes in the result to ensure that the scheduling after scaling is covered by the test, so it's better to use a smaller value here.

Comment (BugenZhao) [reply src/tests/simulation/src/cluster.rs:30]: > Is it OK to apply them for all?

I'm not sure, as the time of e2e tests is reduced with larger barrier interval. https://github.com/risingwavelabs/risingwave/pull/6553#issuecomment-1326080734

Comment (wangrunji0408) [reply src/tests/simulation/src/cluster.rs:30]: Just changed to allow user to specify a config file from argument for e2e test, and use the default config if not specified.

File: Makefile.toml (hunk 2 above)
Comment (wangrunji0408) [reply Makefile.toml:599]: Given that they are all scaling test for now, let's leave it as it is. 🤪

**Comment by wangrunji0408** [comment]
I increased the recovery test to 16 times in CI and 2 of them failed. 🥲

@yezizp2012 Could you help to take a look? 🆘

**Comment by wangrunji0408** [comment]
No error after merging the main branch. Not sure whether it was fixed or just a lucky run.

**Comment by yezizp2012** [comment]
> No error after merging the main branch. Not sure whether it was fixed or just a lucky run.

I guess it was a lucky run. 🥵

**Comment by wangrunji0408** [comment]
Let's take advantage of this luck to get it merged. 🤣

**Comment by yezizp2012** [comment]
I have tried more than 100 times locally and finally reproduce it. 🥵 
image
According to the logs in CI, the test fails due to the panic as follows:
```
thread '<unnamed>' panicked at 'assertion failed: !read_version_vec.is_empty()', src/storage/src/hummock/store/version.rs:323:5
--
  | stack backtrace:
  | 0: rust_begin_unwind
  | at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/std/src/panicking.rs:575:5
```

After discussion with @Li0k , the problem is: the batch request somehow was received when the CN has just been restarted, at which point `read_version_mapping` has not been initialized. We need a mechanism to ensure that we can read the read version based on the table_id, perhaps falling back to read s3.
If the test will also fails due to it in main branch, I think we can merge it firstly and open an issue for it and wait for fix. @wangrunji0408 @Li0k

**Comment by yezizp2012** [comment]
@Li0k found a way to fix it, I will try it locally and if it works we can fix it along with this PR.

File: src/tests/simulation/src/cluster.rs (hunk 1 above)
Comment (yezizp2012) [reply src/tests/simulation/src/cluster.rs:30]: How about only use this config for scale test and allow other tests to use default? The time cost of other simulation test will reduced obviously with larger barrier interval.

Comment (wangrunji0408) [reply src/tests/simulation/src/cluster.rs:30]: Yeah, it is this way now.

**Comment by BugenZhao** [comment]
> How about only use this config for scale test and allow other tests to use default? The time cost of other simulation test will reduced obviously with larger barrier interval.

+1 🤔

**Review by BugenZhao — APPROVED** [review]

File: src/tests/simulation/src/cluster.rs (hunk 1 above)
Comment (yezizp2012) [reply src/tests/simulation/src/cluster.rs:30]: Indeed, that's quite strange that the time cost of recovery test is getting longer, from 4mins+ to 6mins+. 😢

**Comment by yezizp2012** [comment]
> @Li0k found a way to fix it in #6690 , I will try it locally to check if it works.

The panic did not reappear and I think it should be fixed.

**Review by yezizp2012 — APPROVED** [review]

**Pull request merged**